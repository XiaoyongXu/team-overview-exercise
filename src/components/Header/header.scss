.header-root{
  display: flex;
  flex: 1;
  justify-content: center;
  background-color: #11698E;
  padding: 20px;
  .header-main {
    display: flex;
    justify-content: space-between;
    flex-direction: column;
    width: 40%;
    .header-title{
      margin: 0 auto;
      font-style: normal;
      font-weight: bold;
      font-size: 30px;
      line-height: 37px;
      color: #FFFFFF;
    }
  }
}