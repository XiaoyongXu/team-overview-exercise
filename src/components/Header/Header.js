import React from 'react'
import './header.scss'
import Button from '../Button'

const Header = ({
  style,
  children,
  title,
  backLabel,
  onBack = () => {},
}) => {
  return (
    <div className="header-root" data-testid="header" style={style}>
      {backLabel && onBack && <Button
        style={{
          position: 'absolute',
          left: 20,
          top: 20,
        }}
        label={backLabel}
        onClick={onBack}
      />}
      <div className="header-main">
        <div className="header-title">{title}</div>
        {children}
      </div>
    </div>
  )
}

export default Header
