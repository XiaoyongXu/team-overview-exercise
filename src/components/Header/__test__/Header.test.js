import React from 'react'
import ReactDOM from 'react-dom'
import Header from '../Header'
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom'
import renderer from 'react-test-renderer'

afterEach(cleanup)

it("renders without crashing", () => {
  const div = document.createElement("div")
  ReactDOM.render(<Header />, div)
})

it("renders Header correctly with title", () => {
  const { getByTestId } = render(<Header title="test title" />)
  expect(getByTestId('header')).toHaveTextContent("test title")
})

it("renders Header correctly with back button", () => {
  const { getByTestId } = render(<Header backLabel="Back" />)
  expect(getByTestId('header')).toHaveTextContent("Back")
})


it("matches snapshot 1", () => {
  const tree = renderer.create(<Header title="test title" />).toJSON()
  expect(tree).toMatchSnapshot()
})

it("matches snapshot 2", () => {
  const tree = renderer.create(<Header backLabel="Back" />).toJSON()
  expect(tree).toMatchSnapshot()
})