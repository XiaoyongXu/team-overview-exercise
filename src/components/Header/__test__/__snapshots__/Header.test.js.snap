// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1 1`] = `
<div
  className="header-root"
  data-testid="header"
>
  <div
    className="header-main"
  >
    <div
      className="header-title"
    >
      test title
    </div>
  </div>
</div>
`;

exports[`matches snapshot 2 1`] = `
<div
  className="header-root"
  data-testid="header"
>
  <button
    className="btn-icon"
    data-testid="button"
    onClick={[Function]}
    style={
      Object {
        "left": 20,
        "position": "absolute",
        "top": 20,
      }
    }
    type="button"
  >
    Back
  </button>
  <div
    className="header-main"
  >
    <div
      className="header-title"
    />
  </div>
</div>
`;
